storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8381,12 +8381,23 @@ ha_rocksdb::generate_cf_name(const uint index, const TABLE *const table_arg,
     return "";
   }
 
-  // If we didn't find any partitioned/non-partitioned qualifiers, return the
-  // comment itself.  NOTE: this currently handles returning the cf name
-  // specified in the index comment in the case of no partitions, which doesn't
-  // use any qualifiers at the moment. (aka its a special case)
-  if (cf_name.empty() && !key_comment.empty()) {
-    return key_comment;
+  // If we didn't find any partitioned/non-partitioned qualifiers, DO NOT return
+  // the  comment itself. FB Behavior is exactly the opposite which then causes
+  // any index comment to create a new column family.
+
+  // Strip leading and trailing whitespace as that is not part of a valid name
+  if (!cf_name.empty()) {
+    std::string::size_type start = cf_name.find_first_not_of(" ");
+    std::string::size_type end = cf_name.find_last_not_of(" ");
+
+    if (start == std::string::npos && end == std::string::npos) {
+      cf_name.erase();
+    } else {
+      if (end != std::string::npos)
+        cf_name.erase(end + 1);
```

Comment:
This cannot be right, it only erases one character

Reply:
> sequence (1)  string& erase (size_t pos = 0, size_t len = npos)'
> ...
> (1) sequence Erases the portion of the string value that begins at the character position pos and spans len characters (or until the end of the string, if either the content is too short or if len is string::npos. Notice that the default argument erases all characters in the string (like member function clear).

This does raise a good point though, I did not add a specific test case for these parser changes and there should be one.

Reply:
You are right, I confused this with a single arg iterator overload